digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/CalibrationPointDAO.scala

```diff
@@ -51,8 +51,20 @@ object CalibrationPointDAO {
          PROJECT_LINK_ID = $projectLinkId And PROJECT_ID = $projectId And ABS(LINK_M - $segmentMValue) < $epsilon
        """
 
-    Q.queryNA[(Long, Long, Long, Double, Long)](baseQuery).list.map{
-      case (id,linkId, prId, linkM, addressM) => UserDefinedCalibrationPoint(id,linkId, prId, linkM, addressM)
+    Q.queryNA[(Long, Long, Long, Double, Long)](baseQuery).list.map {
+      case (id, linkId, prId, linkM, addressM) => UserDefinedCalibrationPoint(id, linkId, prId, linkM, addressM)
+    }
+  }
+
+  def findEndCalibrationPoint(projectLinkId: Long, projectId: Long): Seq[UserDefinedCalibrationPoint] = {
```

Comment:
Will this work also with reversed project_links? I tried to test this sql in devtest but the only project having reversed links have only one calibration_point per project_link. This might be correct but I don't understand this yet well enough.

SELECT * FROM CALIBRATION_POINT WHERE project_id = 9070648;

Reply:
i will check that project case

Reply:
Might be good to create project where there is a reversed link with multiple calibration points?

Reply:
The Calibration_Point table defines if and where there is one new calibration point defined by the user for one project_link. 
"New endAddressMValue defined by user" logic indicates that the user can only define one new endAddress and not startAddress for one projec_link. (UI should not let user change the startAddress for one link even tough the ALUSSA field in form is not being blocked yet).
However for the query and the recalculation only the endAddress calibration point defined by user matters, thats why we only pick the last one from that project_link.

Reply:
Ok, so this should not be a problem.

Reply:
From this "...Select Max(Address_M)..." I thought that there could be more than one calibration points.